### DISCUSSION TITLE
Hide specific languages from language dropdown

### DISCUSSION THREAD

**Discussion by bjarnef** [umbraco/Umbraco-CMS#14431] [Features and ideas]
Sometimes we prefer to create languages using the languages without country code, e.g. Danish, English, German etc.

However on Umbraco Cloud, Deploy we by default use "English (United States) (en-US)". We can delete this, but when we have previous did this, it may to re-added when creating new environments.

Sometimes one may also want to create a new language, prepare code etc. but not yet ready and one don't want content editors to start working the the new language variant.

Maybe there should be a setting on language or a config setting to hide a specific language from UI.

**Comment by skttl** [comment]
I think the default language (en-US) is a config setting you can change, and by that, prevent it from being autogenerated in new environments.

**Comment by bjarnef** [comment]
@AndyButland do you recall it there is a setting to specify the default language/culture.
I think the Umbraco Cloud support once mentioned it always created `en-US` when a new environment was added, so even though the language is deleted/removed, it may be re-added later.

**Comment by rasmusjp** [comment]
According to the [docs](https://docs.umbraco.com/umbraco-cms/reference/configuration/installdefaultdatasettings) you should be able to change the default data created by updating your appsettings.

While I haven't tested this on Cloud, it should work as long as you have deleted the `en-US` uda file from the Deploy data directory in git.

```json
{
    "Umbraco": {
        "CMS": {
            "InstallDefaultData": { 
                "Languages": {
                    "InstallData": "Values",
                    "Values": ["en", "da"]
                }
            }
        }
    }
}
```

**Reply by bjarnef** [reply]
@rasmusjp thanks for mentioning this. However with Umbraco 11.4.1 and Deploy 11.1.2 I don't see intellisense for `InstallDefaultData`.

**Reply by bjarnef** [reply]
Intellisense suggest `DefaultDataCreation` though, but I don't get any suggestions about `Languages`, `DataTypes` etc.


![image](https://github.com/umbraco/Umbraco-CMS/assets/2919859/17d907a6-68ab-4651-9b52-fbc8458e1558)

**Reply by AndyButland** [reply]
The docs are correct here.  I've raised a [PR](https://github.com/umbraco/Umbraco-CMS/pull/14542) to fix the issue with the generated schema.
